### PR TITLE
Feature/srytal be 37

### PR DESCRIPTION
**Pull request by Ramyathummala**
Implement API to Get all employee details by Super Admin #37

### CODE REVIEW

File: src/controllers/superAdmin/getAllEmployeesBySuperAdminController.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { Request,Response } from "express";
+import { COMMON_ERRORS } from '../../constants/commonErrorMessages';
+import allEmployeesBySuperAdminServices from '../../services/superAdmin/getAllEmployeesBySuperAdminService'
+import { error } from "console";
```
[the remark this answers is not in the document]
Comment (yadagiritalada143) [reply src/controllers/superAdmin/getAllEmployeesBySuperAdminController.ts]: This is resolved.

**Review by yadagiritalada143 — CHANGES_REQUESTED** [review]

File: src/constants/commonErrorMessages.ts [hunk 2]
```diff
@@ -1,7 +1,8 @@
 export const COMMON_ERRORS = {
     USER_FETCHING_ERROR: 'An error occurred while fetching details, Please try again !',
     USER_UPDATING_ERROR: 'An error occurred while updating details, Please try again !',
-    VISITORS_COUNT_UPDATING_ERROR: 'An error occured while updating visitiors count, Please try again !'
+    VISITORS_COUNT_UPDATING_ERROR: 'An error occured while updating visitiors count, Please try again !',
+    FETCHING_ALLEMPLOYEE_ERROR:'An error occurred while fatching allemployeebysuperadmin ! '
```
[the remark this answers is not in the document]
Comment (Ramyathummala) [reply src/constants/commonErrorMessages.ts]: This is resolved

File: src/controllers/superAdmin/getAllEmployeesBySuperAdminController.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ramyathummala) [reply src/controllers/superAdmin/getAllEmployeesBySuperAdminController.ts]: This is resolved

File: src/controllers/superAdmin/getAllEmployeesBySuperAdminController.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    .then(fetchAllEmployeeBySuperAdminResponse => {
+        res.status(200).json(fetchAllEmployeeBySuperAdminResponse);
+    })
+    .catch(error=>{
+        console.error(`Error in fetching all superadmin employee detals:${error} `);
+        res.status(500).json({ success: false, message: COMMON_ERRORS.FETCHING_ALLEMPLOYEE_ERROR });
+        
+    })
+};
+export default {getAllEmployeesBySuperAdmin}
```
[the remark this answers is not in the document]
Comment (Ramyathummala) [reply src/controllers/superAdmin/getAllEmployeesBySuperAdminController.ts]: Reloved this

File: src/services/superAdmin/getAllEmployeesBySuperAdminService.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import UserModel from '../../model/userModel';
+import { error } from 'console';
+
+// interface FetchUserResponse {
```
[the remark this answers is not in the document]
Comment (Ramyathummala) [reply src/services/superAdmin/getAllEmployeesBySuperAdminService.ts]: Resolved this

File: src/routes/superAdminRoutes.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import express, { Router } from 'express';
+import getAllEmployeesBySuperAdminController from '../controllers/superAdmin/getAllEmployeesBySuperAdminController'
+
+
+const superAdminRouter: Router = express.Router();
+
+superAdminRouter.get('/getAllEmployeesBySuperAdmin',getAllEmployeesBySuperAdminController.getAllEmployeesBySuperAdmin);
+
```
[the remark this answers is not in the document]
Comment (Ramyathummala) [reply src/routes/superAdminRoutes.ts]: This is resolved

File: src/controllers/superAdmin/getAllEmployeesBySuperAdminController.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import { Request,Response } from "express";
+import { COMMON_ERRORS } from '../../constants/commonErrorMessages';
+import allEmployeesBySuperAdminServices from '../../services/superAdmin/getAllEmployeesBySuperAdminService'
+import { error } from "console";
+
+const getAllEmployeesBySuperAdmin=(req:Request,res:Response)=>{
```
[the remark this answers is not in the document]
Comment (Ramyathummala) [reply src/controllers/superAdmin/getAllEmployeesBySuperAdminController.ts]: Resolved this

File: src/controllers/superAdmin/getAllEmployeesBySuperAdminController.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ramyathummala) [reply src/controllers/superAdmin/getAllEmployeesBySuperAdminController.ts]: This is resolved

**Pull request merged**